src/SortMsg.chpl

```diff
@@ -29,176 +30,109 @@ module SortMsg
+          var slice = makeDistArray(d.dim(axis).size, t);
+          forall i in d.dim(axis) with (var perpIdx = idx) {
+            perpIdx[axis] = i;
+            slice[i] = array[perpIdx];
           }
-        }
 
-        st.addEntry(rname, createSymEntry(sorted));
-        const repMsg = "created " + st.attrib(rname);
-        sortLogger.debug(getModuleName(),pn,getLineNumber(),repMsg);
-        return new MsgTuple(repMsg, MsgType.NORMAL);
-      }
+          ArgSortMsg.dynamicTwoArrayRadixSort(slice, comparator=myDefaultComparator);
 
-      select gEnt.dtype {
-        when DType.Int64 do return doSort(int);
-        when DType.UInt64 do return doSort(uint);
-        when DType.Float64 do return doSort(real);
-        otherwise {
-          var errorMsg = notImplementedError(pn,gEnt.dtype);
-          sortLogger.error(getModuleName(),pn,getLineNumber(), errorMsg);
-          return new MsgTuple(errorMsg, MsgType.ERROR);
+          forall i in d.dim(axis) with (var perpIdx = idx) {
+            perpIdx[axis] = i;
+            sorted[perpIdx] = slice[i];
+          }
+        }
+      } else {
+        // TODO: make a version of radixSortLSD_keys that does the sort on
+        // slices of `e.a` directly instead of requiring a copy for each slice
+        for idx in DD {
+          const sliceDom = domOnAxis(d, idx, axis),
+                sliced1D = removeDegenRanks(array[sliceDom], 1),
+                sliceSorted = radixSortLSD_keys(sliced1D);
+
+          forall i in sliceDom do sorted[i] = sliceSorted[i[axis]];
         }
       }
+
+      return sorted;
     }
 
-    // https://data-apis.org/array-api/latest/API_specification/generated/array_api.searchsorted.html#array_api.searchsorted
-    @arkouda.registerND
-    proc searchSortedMsg(cmd: string, msgArgs: borrowed MessageArgs, st: borrowed SymTab, param nd: int): MsgTuple throws {
-      param pn = Reflection.getRoutineName();
-      const x1 = msgArgs.getValueOf("x1"),
-            x2 = msgArgs.getValueOf("x2"),
-            side = msgArgs.getValueOf("side"),
-            rname = st.nextName();
 
-      var gEntX1: borrowed GenSymEntry = getGenericTypedArrayEntry(x1, st),
-          gEntX2: borrowed GenSymEntry = getGenericTypedArrayEntry(x2, st);
+    proc sort(array: [?d] ?t, alg: string, axis: int): [d] t throws 
+    where ((t != real) && (t!=int) && (t!=uint(64))) {
+      throw new Error("sort does not support type %s".format(type2str(t)));
+    }
 
-      if side != "left" && side != "right" {
-        throw getErrorWithContext(
-          msg="Unrecognized side: %s".format(side),
-          lineNumber=getLineNumber(),
-          pn,
-          moduleName=getModuleName(),
-          errorClass="NotImplementedError"
-        );
-      }
+    // https://data-apis.org/array-api/latest/API_specification/generated/array_api.searchsorted.html#array_api.searchsorted
+    @arkouda.registerCommand
+    proc searchSorted(x1: [?d1] ?t, x2: [?d2] ?t2, side: string): [d2] int throws
```

Comment:
I think the arguments can be `x1: [?d1] real, x2: [?d2]: real` to avoid instantiating this proc for all the other unused types.

Reply:
(If that doesn't work I'd consider it a bug with `registerCommand`, which I'd be happy to fix).

Reply:
It's not working when I try that: 

```
/home/amandapotts/git/arkouda//src/registry/Commands.chpl:5678: In function 'ark_reg_searchSorted_generic':
/home/amandapotts/git/arkouda//src/registry/Commands.chpl:5684: error: unresolved call 'SortMsg.searchSorted([domain(1,int(64),one)] int(64), [domain(1,int(64),one)] int(64), string)'
/home/amandapotts/git/arkouda//src/SortMsg.chpl:103: note: this candidate did not match: searchSorted(x1: [?d1] real, x2: [?d2] real, side: string)
/home/amandapotts/git/arkouda//src/SortMsg.chpl:103: note: because where clause evaluated to false
/home/amandapotts/git/arkouda//src/registry/Commands.chpl:5684: note: other candidates are:
/home/amandapotts/git/arkouda//src/SortMsg.chpl:128: note:   searchSorted(x1: [?d1] real, x2: [?d2] real, side: string)
  /home/amandapotts/git/arkouda//src/registry/Commands.chpl:5691: called as ark_reg_searchSorted_generic(cmd: string, msgArgs: borrowed MessageArgs, st: borrowed SymTab, type array_dtype_0 = int(64), param array_nd_0 = 1, type array_dtype_1 = int(64), param array_nd_1 = 1)
note: generic instantiations are underlined in the above callstack
make: *** [Makefile:397: arkouda_server] Error 1

```
because it does try to instantiate all the types:
```
proc ark_searchSorted_int_1_int_1(cmd: string, msgArgs: borrowed MessageArgs, st: borrowed SymTab): MsgTuple throws do
  return ark_reg_searchSorted_generic(cmd, msgArgs, st, array_dtype_0=int, array_nd_0=1, array_dtype_1=int, array_nd_1=1);
registerFunction('searchSorted<int64,1,int64,1>', ark_searchSorted_int_1_int_1, 'SortMsg', 103);
```

If you have time to fix it, it would be very beneficial.

Reply:
Okay, that's unfortunate. I'll try to fix it this afternoon (if not, tomorrow).

Reply:
https://github.com/Bears-R-Us/arkouda/pull/3786 should fix it

Reply:
OK, I made the change.  Thanks for the fix.